### PR TITLE
Add release note for kubedb-2024.12.18

### PR DESCRIPTION
**Pull request by pritamdas99**

### CODE REVIEW

**Review by tapojit047 — COMMENTED** [review]
The order of the the databases should be sorted alphabetically.

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]
Lets make it 2024.12.18

File: content/post/kubedb-v2024.12.18/index.md [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+- singlestore
+- solr
+- tls
+- zookeeper
+---
+
+KubeDB **v2024.12.19** is now available! This latest release brings significant performance enhancements, improved reliability, and new features to database management experience on Kubernetes. Here are some of the key features to mention
+- **OpsRequest Support**: New `OpsRequest` features have been added for `Memcached`, `MySQL`, offering greater flexibility for managing database administrative tasks. Moreover, a new `OpsRequest` feature named `ReplicationModeTransformation` has been introduced in this release.
+- **Recommendation Engine**: Recommendation support for `KubeDB` managed kafka has been added.
+- **New Version Support**: New versions have been added for `Elasticsearch`, `MySQL`, `Redis`, `Solr`, `Singlestore`.
```
Comment (tapojit047) [inline content/post/kubedb-v2024.12.18/index.md:48]: `Druid` is missing

**Review by tapojit047 — CHANGES_REQUESTED** [review]

File: content/post/kubedb-v2024.12.18/index.md [hunk 2]
```diff
@@ -0,0 +1,824 @@
...
+    name: memcd-quickstart
+  authentication:
+    secretRef:
+      name: mc-new-auth
+```
+
+Finally, the operator will update the database cluster with the new credential and the old credentials will be stored in the secret with keys username.prev and password.prev.
+We have added a field `.spec.authSecret.activeFrom` to the db yaml which refers to the timestamp of the credential is active from. We also add an annotation kubedb.com/auth-active-from in currently using auth secret which refer to the active from time of this secret.
+
+## MSSQLSERVER
```
Comment (tapojit047) [inline content/post/kubedb-v2024.12.18/index.md:471]: This should be lowercase

**Review by tapojit047 — CHANGES_REQUESTED** [review]

File: content/post/kubedb-v2024.12.18/index.md (hunk 1 above)
Comment (pritamdas99) [reply content/post/kubedb-v2024.12.18/index.md]: added

File: content/post/kubedb-v2024.12.18/index.md [hunk 3]
```diff
@@ -0,0 +1,863 @@
...
+- restore
+- s3
+- security
+- singlestore
+- solr
+- tls
+- zookeeper
+---
+
+KubeDB **v2024.12.19** is now available! This latest release brings significant performance enhancements, improved reliability, and new features to database management experience on Kubernetes. Here are some of the key features to mention
```
Comment (Neaj-Morshad-101) [inline content/post/kubedb-v2024.12.18/index.md:45]: v2024.12.18

File: content/post/kubedb-v2024.12.18/index.md [hunk 4]
```diff
@@ -0,0 +1,863 @@
...
+- **OpsRequest Support**: New `OpsRequest` features have been added for `Memcached`, `MySQL`, offering greater flexibility for managing database administrative tasks. Moreover, a new `OpsRequest` feature named `ReplicationModeTransformation` has been introduced in this release.
+- **Recommendation Engine**: Recommendation support for `KubeDB` managed kafka has been added.
+- **New Version Support**: New versions have been added for `Druid`, `Elasticsearch`, `FerretDB`, `Kafka`, `MariaDB`, `Memcached`, `MSSQLServer`, `MySQL`, `RabbitMQ`, `Redis`, `Solr`, `Singlestore`.
```
Comment (Neaj-Morshad-101) [inline content/post/kubedb-v2024.12.18/index.md:48]: `MSSQLServer` -> `Microsoft SQL Server`

File: content/post/kubedb-v2024.12.18/index.md [hunk 5]
```diff
@@ -0,0 +1,863 @@
...
+- s3
+- security
+- singlestore
+- solr
+- tls
+- zookeeper
+---
+
+KubeDB **v2024.12.19** is now available! This latest release brings significant performance enhancements, improved reliability, and new features to database management experience on Kubernetes. Here are some of the key features to mention
+- **OpsRequest Support**: New `OpsRequest` features have been added for `Memcached`, `MySQL`, offering greater flexibility for managing database administrative tasks. Moreover, a new `OpsRequest` feature named `ReplicationModeTransformation` has been introduced in this release.
```
Comment (Neaj-Morshad-101) [inline content/post/kubedb-v2024.12.18/index.md:46]: Include Microsoft SQL Server

File: content/post/kubedb-v2024.12.18/index.md [hunk 6]
```diff
@@ -0,0 +1,863 @@
...
+- **Recommendation Engine**: Recommendation support for `KubeDB` managed kafka has been added.
+- **New Version Support**: New versions have been added for `Druid`, `Elasticsearch`, `FerretDB`, `Kafka`, `MariaDB`, `Memcached`, `MSSQLServer`, `MySQL`, `RabbitMQ`, `Redis`, `Solr`, `Singlestore`.
+- **Archiver**: Archiver and point-in-time-recovery within `KubeDB` has been enhanced for `MongoDBArchiver`, `MariaDBArchiver`, `MySQLArchiver`, `PostgressArchiver`, `MSSQLArchiver`.
```
Comment (Neaj-Morshad-101) [inline content/post/kubedb-v2024.12.18/index.md:49]: `MSSQLArchiver` -> `MSSQLServerArchiver`

File: content/post/kubedb-v2024.12.18/index.md [hunk 7]
```diff
@@ -0,0 +1,863 @@
...
+
+## Archiver
+Archiver support has been enhanced for `MongoDBArchiver`, `MariaDBArchiver`, `MySQLArchiver`, `PostgressArchiver` and `MSSQLArchiver`. We have replaced the field `spec.walBackup` in spec with `spec.logBackup` with some enhancement. Besides two existing field `RuntimeSettings` and `ConfigSecret`, two more fields have been added and those are:
```
Comment (Neaj-Morshad-101) [inline content/post/kubedb-v2024.12.18/index.md:52]: MSSQLArchiver -> MSSQLServerArchiver

File: content/post/kubedb-v2024.12.18/index.md [hunk 8]
```diff
@@ -0,0 +1,863 @@
...
+    sessionHistoryLimit: 2
+  backupStorage:
+    ref:
+      name: s3-storage
+      namespace: demo
+```
+
+## Recommendation Engine
+
+In this Release, we are introducing `Recommendation` support for KubeDB managed Kafka instances. KubeDB ops-manager generates three types of recommendations for Kafka - Version Update Recommendation, TLS Certificates Rotation Recommendation and Authentication Secret Rotation Recommendation. Authentication Secret rotation is a new type of recommendation supported by kubedb. It recommends to rotate authentication secret of a particular db if only one month remaining for rotating or two third of it's lifespan has been completed.
```
Comment (Neaj-Morshad-101) [inline content/post/kubedb-v2024.12.18/index.md:104]: kubedb -> KubeDB

**Review by Neaj-Morshad-101 — CHANGES_REQUESTED** [review]

**Review by sayedppqq — APPROVED** [review]
LGTM

**Review by AshrafulHaqueToni — COMMENTED** [review]
LGTM!

**Review by tapojit047 — DISMISSED** [review]
LGTM!

**Review by obaydullahmhs — APPROVED** [review]
LGTM!

**Review by raihankhan — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**